### PR TITLE
Add rake task to register feature flags from yml

### PR DESCRIPTION
**Pull request by mrodriguez90**
#### Board:
* [Ticket](https://www.notion.so/rootstrap/Add-a-rake-task-to-register-feature-flags-from-a-centralized-YML-file-62dfea9091264d6aacdc6a609afa1acd?pvs=4)

#### Description:
Add a rake task to register feature flags from a centralized YML file on every release.

#### Notes:
This PR adds a YML file to register feature flags, this YML will be used on every release to register the feature flags in Flipper/active_record. 
This PR also includes descriptions for feature flags, and a bin script to run the rake task that registers the FFs.

#### Preview:
Screen Shot 2023-08-14 at 13 14 59

### CODE REVIEW

File: bin/release.sh [hunk 1]
```diff
@@ -0,0 +1,2 @@
+rake db:migrate
```
Comment (JulianPasquale) [inline bin/release.sh:1]:
```suggestion
#!/usr/bin/env bash

rake db:migrate
```

File: config/feature-flags.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# feature_flag_test:
```
Comment (santib) [inline config/feature-flags.yml:1]: ohh why did you remove the "schema" definition? It was cool!

File: bin/release.sh (hunk 1 above)
Comment (santib) [inline bin/release.sh:1]: you need to do `chmod +x ./bin/release.sh` git add/commit/push, so that this file is executable

File: lib/tasks/feature_flags.rake [hunk 3]
```diff
@@ -0,0 +1,12 @@
+FEATURES_YML_PATH = 'config/feature-flags.yml'.freeze
```
Comment (santib) [inline lib/tasks/feature_flags.rake:1]: `/Users/user/dev/rails_api_base/config/initializers/flipper.rb:1: warning: already initialized constant FEATURES_YML_PATH`

dropping constants like this is a problem, they get defined in the global namespace and are even conflicting between themselves. I'd just use the string directly in this case

File: bin/release.sh (hunk 1 above)
Comment (mrodriguez90) [reply bin/release.sh]: Good call @JulianPasquale

File: config/feature-flags.yml (hunk 2 above)
Comment (mrodriguez90) [reply config/feature-flags.yml:6]: @santib not sure why 😃 , I've added it back again

File: bin/release.sh (hunk 1 above)
Comment (mrodriguez90) [reply bin/release.sh]: thanks!

File: lib/tasks/feature_flags.rake (hunk 3 above)
Comment (mrodriguez90) [reply lib/tasks/feature_flags.rake]: good catch

File: spec/lib/tasks/feature_flags_spec.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
+require 'rake'
+
+RSpec.describe 'Feature Flags', :rake do
+  before(:all) do
+    task_defined = Rake::Task.task_defined?('feature_flags:initialize')
```
Comment (santib) [inline spec/lib/tasks/feature_flags_spec.rb:5]: why wouldn't it be defined?

Comment (mrodriguez90) [reply spec/lib/tasks/feature_flags_spec.rb:5]: @santib I'm not sure but when I run the test without this `before` block I get the following error:
Screen Shot 2023-08-18 at 14 04 42

Also, I found it to be common practice to do this in these kinds of tests.

File: spec/lib/tasks/feature_flags_spec.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
+require 'rake'
+
+RSpec.describe 'Feature Flags', :rake do
```
Comment (santib) [inline spec/lib/tasks/feature_flags_spec.rb:3]: to avoid testing rake tasks, I usually take the approach of moving the ruby code to a ruby class. Something like

```rb
# lib/tasks/feature_flags.rake
require 'lib/tasks/feature_flags'

namespace :feature_flags do
  desc 'Regiser new feature flags from config/feature-flags.yml'
  task initialize: :environment do
    Tasks::FeatureFlags.new.init_all
  end
end
```

```rb
# lib/tasks/feature_flags.rb
module Tasks
  class FeatureFlags
    def init_all
      YAML.load_file('config/feature-flags.yml', fallback: {}).each do |key, _options|
        next if Flipper.exist?(key)

        Flipper.add(key)
      end
    end
  end
end
```

and then you just test the ruby class. BTW if doing this approach you can go back to use constants defined under the corresponding Ruby module/class

File: lib/tasks/feature_flags.rake [hunk 6]
```diff
@@ -0,0 +1,10 @@
+namespace :feature_flags do
+  desc 'Regiser new feature flags from config/feature-flags.yml'
```
Comment (santib) [inline lib/tasks/feature_flags.rake:2]:
```suggestion
  desc 'Register new feature flags from config/feature-flags.yml'
```

File: bin/release.sh [hunk 7]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
+
+rake db: migrate
+rake feature_flags: initialize
```
Comment (santib) [inline bin/release.sh:4]: not related to this PR but I usually recommend running all rails/rake commands prepended with `bundle exec`

https://thoughtbot.com/blog/but-i-dont-want-to-bundle-exec

**Review by santib — COMMENTED** [review]
Great 👏

**Review by santib — APPROVED** [review]

**Review by JulianPasquale — APPROVED** [review]
